src/core/bob_client.rs

```diff
@@ -194,31 +195,20 @@ impl BobClient {
             .then(move |client_res| match client_res {
                 Ok(mut cl) => cl
                     .ping(Request::new(Null {}))
-                    .timeout(to)
                     .map(move |_| ClusterResult {
                         node: n1,
                         result: BackendPingResult {},
                     })
-                    .map_err(move |e| ClusterResult {
-                        node: n2.clone(),
-                        result: {
-                            if e.is_elapsed() {
-                                Error::Timeout
-                            } else if e.is_timer() {
-                                panic!("Timeout can't failed in core - can't continue")
-                            } else {
-                                let err = e.into_inner();
-                                Error::Failed(format!(
-                                    "Ping operation for {} failed: {:?}",
-                                    n2, err
-                                ))
-                            }
-                        },
-                    }),
+                    .map_err(|e| Error::StorageError(format!("ping operation error: {}", e))),
                 Err(_) => panic!("Timeout failed in core - can't continue"), //TODO
             })
             .compat()
             .boxed()
+            .timeout(to)
```

Comment:
Does this one chaining works for checking timeout after making `map` and `map_err`?